sdgx/data_processors/transformers/fixed_combination.py

```diff
@@ -15,41 +15,89 @@ class FixedCombinationTransformer(Transformer):
+    |  2   |  B   |   1001   |  Apple   |   20   |   20   |
+    """
 
-    Attributes:
-        fixed_combinations (dict[str, set[str]]): A dictionary mapping column names to sets of column names that have fixed relationships with them.
+    fixed_combinations: dict[str, set[str]]
+    """
+    A dictionary mapping column names to sets of column names that have fixed relationships with them.
     """
 
-    def __init__(self):
-        super().__init__()  # Call the parent class's initialization method
-        # Initialize the variable in the initialization method
+    simplified_fixed_combinations: dict[str, set[str]]
+    """
+    A dictionary mapping column names to sets of column names that have fixed relationships with them.
+    """
 
-        self.fixed_combinations: dict[str, set[str]] = {}
-        """
-        A dictionary mapping column names to sets of column names that have fixed relationships with them.
-        """
+    column_mappings: dict[(str, str), dict[str, str]]
+    """
+    A dictionary mapping tuples of column names to dictionaries of value mappings.
+    """
 
-        self.simplified_fixed_combinations: dict[str, set[str]] = {}
-        """
-        A dictionary mapping column names to sets of column names that have fixed relationships with them.
-        """
+    is_been_specified: bool
+    """
+    A boolean that flag if exist specific combinations by user.
+    If true, needn't running this auto detect transform.
+    """
 
+    def __init__(self):
+        super().__init__()
+        self.fixed_combinations: dict[str, set[str]] = {}
+        self.simplified_fixed_combinations: dict[str, set[str]] = {}
         self.column_mappings: dict[(str, str), dict[str, str]] = {}
+        self.is_been_specified = False
+
+    @property
+    def is_exist_fixed_combinations(self) -> bool:
         """
-        A dictionary mapping tuples of column names to dictionaries of value mappings.
+        A boolean that flag if inspector have inspected some fixed combinations.
+        If False, needn't running this auto detect transform.
         """
+        return bool(self.fixed_combinations)
 
     def fit(self, metadata: Metadata | None = None, **kwargs: dict[str, Any]):
         """Fit the transformer and save the relationships between columns.
 
         Args:
             metadata (Metadata): Metadata object
         """
+        # Check if exist specific combinations by user. If True, needn't run this auto-detect transform.
+        if metadata.get("specific_combinations"):
+            logger.info(
+                "Fit data using FixedCombinationTransformer(been specified)... Finished (No action)."
+            )
+            self.is_been_specified = True
+            self.fitted = True
+            return
+
+        # Check if exist fixed combinations, if not exist, needn't run this auto-detect transform.
         self.fixed_combinations = metadata.get("fixed_combinations")
```

Comment:
```suggestion
        self.fixed_combinations = metadata.get("fixed_combinations") or dict()
```

as `self.fixed_combinations: dict[str, set[str]]`